openml/_api_calls.py

```diff
@@ -107,7 +132,19 @@ def _download_minio_file(
     if destination.is_file() and not exists_ok:
         raise FileExistsError(f"File already exists in {destination}.")
 
-    client = minio.Minio(endpoint=parsed_url.netloc, secure=False)
+    proxy_client = None
+    if isinstance(proxy, str):
+        proxy_client = ProxyManager(proxy)
+    else:
+        proxy_url = resolve_env_proxies(parsed_url.geturl()) if proxy else None
+        if proxy_url:
+            proxy_client = ProxyManager(proxy_url)
```

Comment:
If we were to change ``proxy`` to be of type `str` and use a default `"auto"`, instead of relying on mixed types, the resulting code would look much nicer (unfortunately I can't make the suggestion in the same batch because of the github web interface). As I understand it, the behavior should still be equivalent.
```suggestion
    if proxy == "auto":
      proxy = resolve_env_proxies(parsed_url.geturl())
    proxy_client = ProxyManager(proxy) if proxy else None
```